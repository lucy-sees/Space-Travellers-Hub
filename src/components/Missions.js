const Missions = () => <h2>Missions page</h2>;

export default Missions;
