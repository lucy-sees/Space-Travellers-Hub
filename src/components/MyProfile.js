const myProfile = () => <h2>My Profile page</h2>;

export default myProfile;
