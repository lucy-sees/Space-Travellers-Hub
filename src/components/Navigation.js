import '../styles/Navigation.css';
import logo from '../assets/planet.png';

const Navigation = () => (
  <nav className="navmenu-desktop">
    <img src={logo} alt="logo" className="logo-img" />
    <h1 className="logo">Space Travellers&apos; Hub</h1>
    <ul className="nav-items">
      <li className="nav-item">1</li>
      <li className="nav-item">2</li>
      <li className="nav-item">3</li>
    </ul>
  </nav>
);

export default Navigation;
