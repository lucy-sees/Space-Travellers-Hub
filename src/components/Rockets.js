const Rockets = () => <h2>Rockets page</h2>;

export default Rockets;
