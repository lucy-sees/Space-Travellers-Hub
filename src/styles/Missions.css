.missions {
  color: #333;
  border: 1px solid #ddd;
  border-collapse: collapse;
  margin: 20px 5%;
}

.missions td {
  border: 1px solid #ddd;
  padding: 5px;
}

.missions th {
  border: 1px solid #ddd;
  text-align: left;
  padding: 5px;
  font-size: 20px;
  font-weight: 600;
}

.td-mission {
  vertical-align: top;
  width: 10%;
  font-size: 20px;
  font-weight: 600;
}

.status,
.button {
  text-align: center;
  width: 10%;
}
