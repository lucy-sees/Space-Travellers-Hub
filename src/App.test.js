import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Navigation from './components/Navigation';
import MyProfile from './components/MyProfile';
import Rockets from './components/Rockets';
import Missions from './components/Missions';

const mockStore = configureStore([thunk]);

describe('Navigation Component', () => {
  test('renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Profile Component', () => {
  test('renders correctly', () => {
    const store = mockStore({
      missions: {
        missions: [
          { id: 1, name: 'Mission 1' },
          { id: 2, name: 'Mission 2' },
        ],
        joinedMissions: [1],
      },
      rockets: {
        data: [
          { id: 1, name: 'Rocket 1' },
          { id: 2, name: 'Rocket 2' },
        ],
        joinedRockets: [2],
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <MyProfile />
        </MemoryRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Rockets Component', () => {
  test('matches snapshot', () => {
    const store = mockStore({
      rockets: {
        isLoading: false,
        data: [
          {
            id: 1,
            name: 'Rocket 1',
            description: 'Description 1',
            image: 'rocket1.jpg',
          },
          {
            id: 2,
            name: 'Rocket 2',
            description: 'Description 2',
            image: 'rocket2.jpg',
          },
        ],
        joinedRockets: [1],
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Missions Component', () => {
  test('renders correctly', () => {
    const store = mockStore({
      missions: {
        isLoading: false,
        missions: [
          { id: 1, name: 'Mission 1', description: 'Description 1' },
          { id: 2, name: 'Mission 2', description: 'Description 2' },
        ],
        joinedMissions: [1],
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
