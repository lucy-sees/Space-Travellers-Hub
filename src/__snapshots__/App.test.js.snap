// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component renders correctly 1`] = `
<DocumentFragment>
  <table
    class="missions"
  >
    <thead>
      <tr>
        <th
          class="mission"
        >
          Mission
        </th>
        <th
          class="description"
        >
          Description
        </th>
        <th
          class="status"
        >
          Status
        </th>
        <th
          class="button"
        >
          Â 
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="rows"
      >
        <td
          class="td-mission"
        >
          Mission 1
        </td>
        <td
          class="description td-description"
        >
          Description 1
        </td>
        <td
          class="status"
        >
          <span
            class="mission-status-leave"
          >
            Active Member
          </span>
        </td>
        <td
          class="button"
        >
          <button
            class="td-leave-mission"
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
      <tr
        class="join-rows"
      >
        <td
          class="td-mission"
        >
          Mission 2
        </td>
        <td
          class="description td-description"
        >
          Description 2
        </td>
        <td
          class="status"
        >
          <span
            class="mission-status-join"
          >
            NOT A MEMBER
          </span>
        </td>
        <td
          class="button"
        >
          <button
            class="td-join-mission"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Navigation Component renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navmenu-desktop"
  >
    <img
      alt="logo"
      class="logo-img"
      src="planet.png"
    />
    <h1
      class="logo"
    >
      Space Travellers' Hub
    </h1>
    <ul
      class="nav-items"
    >
      <li
        class="nav-item"
      >
        <a
          aria-current="page"
          class="activeLink"
          href="/"
        />
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nonActiveLink"
          href="/rockets"
        >
          Rockets
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nonActiveLink"
          href="/missions"
        >
          Missions
        </a>
      </li>
      <li
        class="nav-divider"
      >
        |
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nonActiveLink"
          href="/my-profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Profile Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="profile"
  >
    <table
      class="profile-missions"
    >
      <thead>
        <tr>
          <th>
            My Missions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Mission 1
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="profile-rockets"
    >
      <thead>
        <tr>
          <th>
            My Rockets
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Rocket 2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Rockets Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rockets-container"
  >
    <div
      class="rocket-left"
    >
      <img
        alt="Rocket 1"
        class="rocket-image"
        src="rocket1.jpg"
      />
      <div
        class="rocket-details"
      >
        <h2
          class="rocket-name"
        >
          Rocket 1
        </h2>
        <p
          class="rocket-desc"
        >
           
          <span
            class="reserved-badge"
          >
            Reserved
          </span>
          Description 1
        </p>
        <button
          class="rocket-btn cancel"
          data-testid="cancel-reservation-button-1"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
    <div
      class="rocket-left"
    >
      <img
        alt="Rocket 2"
        class="rocket-image"
        src="rocket2.jpg"
      />
      <div
        class="rocket-details"
      >
        <h2
          class="rocket-name"
        >
          Rocket 2
        </h2>
        <p
          class="rocket-desc"
        >
          Description 2
        </p>
        <button
          class="rocket-btn"
          data-testid="reserve-button-2"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
